client/my-sites/upgrades/domain-management/dns/email-provider.jsx

```diff
@@ -18,33 +17,36 @@ import FormTextInput from 'components/forms/form-text-input';
 import notices from 'notices';
 import * as upgradesActions from 'lib/upgrades/actions';
 
-class Office365 extends Component {
+class EmailProvider extends Component {
 	constructor( props ) {
 		super( props );
 		this.state = { token: '', submitting: false };
 	}
 
 	onChange = ( event ) => {
 		const { value } = event.target;
-		this.setState( { token: value } );
+		this.setState( { token: trim( value ) } );
 	}
 
 	onAddDnsRecords = ( event ) => {
 		event.preventDefault();
 		this.setState( { submitting: true } );
 
-		const { domain, translate } = this.props,
-			variables = {
-				token: this.state.token,
-				domain,
-				mxdata: replace( domain, '.', '-' ) + '.mail.protection.outlook.com.'
-			};
+		const { domain, translate, template } = this.props;
+		let variables = {
+			token: this.state.token,
+			domain
+		};
 
-		upgradesActions.applyDnsTemplate( domain, dnsTemplates.MICROSOFT_OFFICE365, variables, ( error ) => {
+		if ( template.modifyVariables ) {
+			variables = template.modifyVariables( variables );
+		}
+
+		upgradesActions.applyDnsTemplate( domain, template.dnsTemplate, variables, ( error ) => {
 			if ( error ) {
-				notices.error( error.message || translate( 'The DNS records have not been added.' ) );
+				notices.error( error.message || translate( 'The DNS records were not able to be added.' ) );
```

Comment:
Thanks for the feedback!

We've had this PR in review for so long, that I went ahead and merged, but I am going to ask the editorial team to review this and I will update it with whatever suggestions they make.